### PR TITLE
Upgrade rust sdk to .55

### PR DESCRIPTION
**Pull request by julioromano**
Accommodates some breaking API changes.

### CODE REVIEW

**Comment by julioromano** [comment]
Assigning to @ganfra as he's requested this build together with me.

File: libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/roomlist/RoomListExtensions.kt [hunk 1]
```diff
@@ -93,7 +93,11 @@ fun RoomListService.syncIndicator(): Flow<RoomListServiceSyncIndicator> =
             }
         }
         tryOrNull {
-            syncIndicator(listener)
+            syncIndicator(
+                200u, // Same value which was previously hardcoded in the rust-sdk
```
Comment (ganfra) [inline libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/roomlist/RoomListExtensions.kt:97]: Can you use 1000u? we wanted to expand it a bit. Also can we name those constants?

File: libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/room/RustMatrixRoom.kt [hunk 2]
```diff
@@ -187,7 +187,13 @@ class RustMatrixRoom(
         _membersStateFlow.value = MatrixRoomMembersState.Pending(prevRoomMembers = currentMembers)
         var rustMembers: List<RoomMember>? = null
         try {
-            rustMembers = innerRoom.members()
+            rustMembers = buildList {
+                while (true) {
+                    // Loading the whole iterator as a stop-gap measure.
+                    // We should probably implement some sort of paging in the future.
+                    addAll(innerRoom.members().nextChunk(100u) ?: break)
```
Comment (ganfra) [inline libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/room/RustMatrixRoom.kt:194]: Should we put a higher chunk size for now?

File: libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/roomlist/RoomListExtensions.kt (hunk 1 above)
Comment (julioromano) [reply libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/roomlist/RoomListExtensions.kt:97]: OFC!

File: libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/room/RustMatrixRoom.kt (hunk 2 above)
Comment (julioromano) [reply libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/room/RustMatrixRoom.kt:194]: 1000 is ok?

Comment (ganfra) [reply libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/room/RustMatrixRoom.kt]: I guess!

**Review by ganfra — APPROVED** [review]

**Pull request merged**